keps/sig-network/20190125-ingress-api-group.md

```diff
@@ -389,38 +382,100 @@ type IngressSpec struct {
   Class string
   ...
 }
+```
+
+##### Interoperability with previous annotation
+
+The `kubernetes.io/ingress.class` annotation on v1beta1 Ingresses will be used
+to set the value of the `class` field on v1 Ingresses as part of API conversion.
+When both the `class` field and annotation are set, the field will take
```

Comment:
Annotations have to win to deal with pre-v1 implementations.

Reply:
Is that still true if the annotation from pre-v1 versions is automatically being converted to be the field in v1?

Reply:
In general, the older mechanism should take precedence.  That said, because this is coming with a new API version, this could be viable.  We'd have to add validation that totally disallows that annotation on v1 type.  Otherwise it's ambiguous what to do when you write v1 with annotation AND class field set and then read it as v1beta1 - do you leave the annotation (which might differ from class field) or do you change the annotation?  Both are terrible.

If we go this path (conversion) we should just force them to be mutually exclusive.